### PR TITLE
release(wallet-mobile): v4.27.1-rc.1 pool transition fixes

### PR DESCRIPTION
**Pull request by jorbuedo**
##### Ticket

[YOMO-1477 Modal open after delegation
](https://emurgo.atlassian.net/browse/YOMO-1477)
[YOMO-1480 Modal separation
](https://emurgo.atlassian.net/browse/YOMO-1480)
[YOMO-1478 Keep modal open until navigation](https://emurgo.atlassian.net/browse/YOMO-1478)

### CODE REVIEW

**Comment by jorbuedo** [comment]
I think it's ready. Once squashed it should probably be picked to develop as well

File: apps/wallet-mobile/src/Staking/PoolTransition/PoolTransitionModal.tsx [hunk 1]
```diff
@@ -25,9 +26,11 @@ export const PoolTransitionModal = ({
     closeModal()
   }
 
-  const handleOnUpdate = () => {
+  const handleOnUpdate = async () => {
+    setIsLoading(true)
+    await onContinue()
+    setIsLoading(false)
```
Comment (michalszorad) [inline apps/wallet-mobile/src/Staking/PoolTransition/PoolTransitionModal.tsx]:
```suggestion
try {
    setIsLoading(true)
    await onContinue()
    } finally {
     setIsLoading(false)
    }
```

**Review by michalszorad — APPROVED** [review]
Looks good, added 1 comment about `setIsLoading`

File: apps/wallet-mobile/src/Staking/PoolTransition/PoolTransitionModal.tsx (hunk 1 above)
Comment (jorbuedo) [reply apps/wallet-mobile/src/Staking/PoolTransition/PoolTransitionModal.tsx]: navigateToUpdate (which is the function passed as onContinue) already does a try catch, but sure no harm in adding it here as well.

**Pull request merged**